src/components/Dashboard/Tiles/Widgets/WidgetsEditModal.tsx

```diff
@@ -135,70 +136,168 @@ const WidgetOptionTypeSwitch: FC<{
   handleChange: (key: string, value: IntegrationOptionsValueType) => void;
 }> = ({ option, widgetId, propName: key, value, handleChange }) => {
   const { t } = useTranslation([`modules/${widgetId}`, 'common']);
-  const { primaryColor } = useColorTheme();
+  const { colorScheme } = useMantineTheme();
 
   switch (option.type) {
     case 'switch':
+      var info = option.info !== undefined ? option.info : false;
       return (
-        <Switch
-          label={t(`descriptor.settings.${key}.label`)}
-          checked={value as boolean}
-          onChange={(ev) => handleChange(key, ev.currentTarget.checked)}
-        />
+        <Group align="center" spacing="sm">
+          <Switch
+            label={t(`descriptor.settings.${key}.label`)}
+            checked={value as boolean}
+            onChange={(ev) => handleChange(key, ev.currentTarget.checked)}
+            {...option.inputProps}
+          />
+          {info?
+            <Tooltip.Floating
+              label={t(`descriptor.settings.${key}.info`)}
+              position="right-start"
+              c={ colorScheme === 'light' ? "black" : "dark.0" }
+              color={ colorScheme === 'light' ? "dark.0" : "dark.6" }
+              styles={{ tooltip: { '&': { maxWidth: 300, }, }, }}
+              multiline
+            >
+              <IconAlertCircle size="1.25rem" style={{ display: 'block', opacity: 0.5 }} />
+            </Tooltip.Floating> : undefined
+          }
+        </Group>
       );
     case 'text':
+      var info = option.info !== undefined ? option.info : false;
       return (
-        <TextInput
-          color={primaryColor}
-          label={t(`descriptor.settings.${key}.label`)}
-          value={value as string}
-          onChange={(ev) => handleChange(key, ev.currentTarget.value)}
-        />
+        <Stack spacing={0}>
+          <Group align="center" spacing="sm">
+            <Text size="0.875rem" weight="500">{t(`descriptor.settings.${key}.label`)}</Text>
+            {info?
+              <Tooltip.Floating
+                label={t(`descriptor.settings.${key}.info`)}
+                position="right-start"
+                c={ colorScheme === 'light' ? "black" : "dark.0" }
+                color={ colorScheme === 'light' ? "dark.0" : "dark.6" }
+                styles={{ tooltip: { '&': { maxWidth: 300, }, }, }}
```

Comment:
Isn't this redundant?

Reply:
c is text color, color is background color. Mantine abbreviation for you

Reply:
No, I mean the
```
'&': {
  ....
}
```

Isn't that the same as:


```
{
  ....
}
```